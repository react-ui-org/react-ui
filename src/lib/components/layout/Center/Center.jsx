import PropTypes from 'prop-types';
import React from 'react';
import styles from './Center.scss';

export const Center = (props) => (
  <div className={styles.root}>
    <div>
      {props.children}
    </div>
  </div>
);

Center.defaultProps = {
  children: null,
};

Center.propTypes = {
  children: PropTypes.node,
};

export default Center;
