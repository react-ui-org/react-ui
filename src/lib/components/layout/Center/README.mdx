---
name: Center
menu: 'Components: Layouts'
route: /components/layout/center
---

# Center

The Center layout aligns content both horizontally and vertically.

import {
  Playground,
  Props,
} from 'docz'
import { Placeholder } from '../../../../docs/_components/Placeholder/Placeholder'
import { Center } from './Center'

## Basic Usage

To implement the Center component, you need to import it first:

```js
import { Center } from '@react-ui-org/react-ui';
```

And use it:

<Playground>
  <Placeholder height="200px" bordered>
    <Center>
      <Placeholder bordered>
        Look, I&apos;m centered!
      </Placeholder>
    </Center>
  </Placeholder>
</Playground>

See [API](#api) for all available options.

## General Guidelines

- The Center takes up all **available horizontal space and expands vertically**
  according to the length of its content. Depending on your layout, you may need
  to define the height of its parent to see the vertical alignment work.

- The Center layout uses `flexbox` for alignment. You may need to define the
  desired width of your centered content because it will be shrunk to the
  smallest possible width by default.

## API

<Props of={Center} />

---

Next: [FormLayout â†’](/components/layout/form-layout)
