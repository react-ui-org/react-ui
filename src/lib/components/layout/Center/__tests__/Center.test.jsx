import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Center } from '../Center';

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = shallow(<Center />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with children', () => {
    const tree = shallow(<Center><div>test</div></Center>);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
