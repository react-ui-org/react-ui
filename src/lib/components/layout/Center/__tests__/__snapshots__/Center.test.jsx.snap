// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<div
  className="root"
>
  <div />
</div>
`;

exports[`rendering renders correctly with children 1`] = `
<div
  className="root"
>
  <div>
    <div>
      test
    </div>
  </div>
</div>
`;
